crates/database/src/trades.rs

```diff
@@ -106,6 +106,31 @@ AND t.log_index BETWEEN (SELECT * from previous_settlement) AND $2
         .await
 }
 
+pub async fn token_first_trade_block(
+    ex: &mut PgConnection,
+    token: Address,
+) -> Result<Option<i64>, sqlx::Error> {
+    const QUERY: &str = r#"
+SELECT MIN(sub.block_number) AS earliest_block
+FROM (
+    SELECT t.block_number
+    FROM trades t
+    JOIN orders o ON t.order_uid = o.uid
+    WHERE o.sell_token = $1 OR o.buy_token = $1
```

Comment:
Does it maybe speed up the query to sort each subquery by `block_number` and limit to 1? Maybe the optimizer doesn't realize that it should be able to use the index on `trades` to quickly sort by block_number.

Reply:
All the indexes are properly used. Sorting and then limiting has the worst performance. Using `MIN(t.block_number)` slightly improves the execution time(by a couple of %).